src/code/RepositorySettings.cs

```
@@ -166,33 +166,34 @@ public static PSRepositoryInfo Update(string repoName, Uri repoUri, int repoPrio
 
-                // determine if existing repository node (which we wish to update) had Url to Uri attribute
+                // determine if existing repository node (which we wish to update) had Url or Uri attribute
                 Uri thisUrl = null;
-                if (repoUri != null)
+                if (repoUri != null) 
                 {
-                    if (urlAttributeExists)
-                    {
-                        node.Attribute("Url").Value = repoUri.AbsoluteUri;
-                        // Create Uri from node Uri attribute to create PSRepositoryInfo item to return.
-                        if (!Uri.TryCreate(node.Attribute("Url").Value, UriKind.Absolute, out thisUrl))
+                    if (!Uri.TryCreate(repoUri.AbsoluteUri, UriKind.Absolute, out thisUrl))
                         {
                             throw new PSInvalidOperationException(String.Format("Unable to read incorrectly formatted Url for repo {0}", repoName));
                         }
+
+                    if (urlAttributeExists)
+                    {
+                        node.Attribute("Url").Value = thisUrl.AbsoluteUri;
                     }
-                    else
+                    else{
+                        node.Attribute("Uri").Value = thisUrl.AbsoluteUri;
+                    }
+                }
+                else 
+                {
+                    if (urlAttributeExists)
                     {
-                        // Uri attribute exists
-                        node.Attribute("Uri").Value = repoUri.AbsoluteUri;
-                        if (!Uri.TryCreate(node.Attribute("Uri").Value, UriKind.Absolute, out thisUrl))
-                        {
-                            throw new PSInvalidOperationException(String.Format("Unable to read incorrectly formatted Uri for repo {0}", repoName));
-                        }
+                        Uri.TryCreate(node.Attribute("Url").Value, UriKind.Absolute, out thisUrl);
```

I think we should test for failure on Uri creation.  Seems unlikely it would fail, but if it does that means the repository entry is corrupted and useless.  At minimum we should throw an error informing the user.  We might consider just removing the entry, but that may be too drastic.

---

Agreed on not removing the entry, just throwing an error

---

I made the change you requested. I'm not sure if I throw the right kind of error with the right message so I'd really appreciate your input on that.